subtle_training_bugs.py

```
@@ -0,0 +1,787 @@
+    loss = nn.functional.cross_entropy(output, target)
+
+    if apply_updates:
+        optimizer.zero_grad()
+        loss.backward()
+
+    return loss
+
+
+def adaptive_training_step(model, data, target, optimizer, use_gradients):
+    """Adaptive training based on runtime conditions"""
+    loss = process_batch(model, data, target, optimizer, use_gradients)
```

Ruff (0.13.1)

Local variable `loss` is assigned to but never used

Remove assignment to unused variable `loss`

(F841)